ocs_ci/helpers/e2e_helpers.py

```diff
@@ -392,3 +410,335 @@ def validate_mcg_object_expiration(
+
+    # Patch and delete existing bucketclass
+    bc_obj = OCP(
+        kind=constants.BUCKETCLASS, namespace=config.ENV_DATA["cluster_namespace"]
+    )
+    for bc in bc_obj.get()["items"]:
+        assert bc_obj.patch(
+            resource_name=bc["metadata"]["name"],
+            params=params,
+            format_type="merge",
+        ), "Failed to change the parameter in bucketclass"
+        logger.info(f"Deleting bucketclass: {bc['metadata']['name']}")
+        bc_obj.delete(resource_name=bc["metadata"]["name"])
+
+    # Delete noobaa secrets
+    logger.info("Deleting noobaa related secrets")
+    if is_kms_enabled():
+        dep_ocp.exec_oc_cmd(
+            "delete secrets noobaa-admin noobaa-endpoints noobaa-operator noobaa-server"
+        )
+    else:
+        dep_ocp.exec_oc_cmd(
+            "delete secrets noobaa-admin noobaa-endpoints noobaa-operator "
+            "noobaa-server noobaa-root-master-key-backend noobaa-root-master-key-volume"
+        )
+
+    # Scale back noobaa-operator deployment
+    logger.info(
+        f"Scaling back {constants.NOOBAA_OPERATOR_DEPLOYMENT} deployment to replica: 1"
+    )
+    dep_ocp.exec_oc_cmd(
+        f"scale deployment {constants.NOOBAA_OPERATOR_DEPLOYMENT} --replicas=1"
+    )
+
+    # Wait and validate noobaa PVC is in bound state
+    pvc_obj.wait_for_resource(
+        condition=constants.STATUS_BOUND,
+        resource_name=noobaa_pvc_obj[0].name,
+        timeout=600,
+        sleep=120,
+    )
+
+    # Validate noobaa pods are up and running
+    pod_obj = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+    noobaa_pods = get_noobaa_pods()
+    pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        resource_count=len(noobaa_pods),
+        selector=constants.NOOBAA_APP_LABEL,
+        timeout=900,
+    )
+
+    # Verify everything running fine
+    logger.info("Verifying all resources are Running and matches expected result")
+    self.sanity_helpers.health_check(tries=120)
+
+    # Since the rebuild changed the noobaa-admin secret, update
+    # the s3 credentials in mcg_object_session
+    mcg_obj_session.update_s3_creds()
+
+    # Verify default backingstore/bucketclass
+    default_bs = OCP(
+        kind=constants.BACKINGSTORE, namespace=config.ENV_DATA["cluster_namespace"]
+    ).get(resource_name=DEFAULT_NOOBAA_BACKINGSTORE)
+    default_bc = OCP(
+        kind=constants.BUCKETCLASS, namespace=config.ENV_DATA["cluster_namespace"]
+    ).get(resource_name=DEFAULT_NOOBAA_BUCKETCLASS)
+    assert (
+        default_bs["status"]["phase"]
+        == default_bc["status"]["phase"]
+        == constants.STATUS_READY
+    ), "Failed: Default bs/bc are not in ready state"
+
+    # Create OBCs
+    logger.info("Creating OBCs after noobaa rebuild")
+    bucket_factory_session(amount=3, interface="OC", verify_health=True)
+
+
+def validate_noobaa_db_backup_recovery_locally_system(
```

Comment:
same as my above comment, please covert this to a fixture

Reply:
done